cmd/mkroot/commands.go

```diff
@@ -340,15 +330,51 @@ func Exists(file string) bool {
 	return true
 }
 
-func Mkfs() (*mkfs, error) {
-	if !Exists(fallocatePath) {
-		return nil, fmt.Errorf("could not find '%s'", fallocatePath)
+var (
+	mkfsPath      = "/sbin/mkfs"
+	mountPath     = "/usr/bin/mount"
+	fallocatePath = "/usr/bin/fallocate"
+	umountPath    = "/usr/bin/umount"
+)
+
+func findExecutable(exe string) (string, error) {
+	if Exists(exe) {
+		return exe, nil
 	}
-	if !Exists(mountPath) {
-		return nil, fmt.Errorf("could not find '%s'", mountPath)
+	_, file := filepath.Split(exe)
+	exePath, err := exec.LookPath(file)
+	if err != nil {
+		return "", fmt.Errorf("%s not found: %w", file, err)
 	}
+	return exePath, nil
+}
+
+func Mkfs(name, size, fstype string) (*mkfs, error) {
```

Comment:
this function takes care of some of the heavy lifting now. we don't pass `name`, `size` and `fstype` through optional functions anymore.